File: HRM/HRM/Models/Cores/Position.cs
using System;
using HRM.Models.Bases;

namespace HRM.Models.Cores
{
    public class Position : BaseModel
    {
        public string PositionId { get; set; }
        public string PositionName { get; set; }
        public string Description { get; set; }
        public double? BaseMonthSalary { get; set; }
        public double? BaseHourSalary { get; set; }
        public double? BaseOtSalaryPerHour { get; set; }
        public int? BaseDateOff { get; set; }
        
        public double? BaseLateMoney { get; set; }
    }
}

File: front-end/src/api/position.js
import request from '@/utils/request';

export function getAllPosition() {
  return request({
    url: '/api/positions',
    method: 'get'
  });
}
export function getPosition(positionId) {
  return request({
    url: `/api/positions/${positionId}`,
    method: 'get'
  });
}
export function updatePosition(positionId, dataParam) {
  const data = {
    PositionName: dataParam.positionName,
    Description: dataParam.description,
    BaseMonthSalary: dataParam.baseMonthSalary,
    BaseHourSalary: dataParam.baseHourSalary,
    BaseOtSalaryPerHour: dataParam.baseOtSalaryPerHour,
    BaseDateOff: dataParam.baseDateOff,
    BaseLateMoney: dataParam.baseLateMoney
  };
  return request({
    url: `/api/positions/${positionId}`,
    method: 'patch',
    data
  });
}
export function createPosition(dataParam) {
  const data = {
    PositionName: dataParam.positionName,
    Description: dataParam.description,
    BaseMonthSalary: dataParam.baseMonthSalary,
    BaseHourSalary: dataParam.baseHourSalary,
    BaseOtSalaryPerHour: dataParam.baseOtSalaryPerHour,
    BaseDateOff: dataParam.baseDateOff,
    BaseLateMoney: dataParam.baseLateMoney
  };

  return request({
    url: `/api/positions`,
    method: 'post',
    data
  });
}
export function deletePosition(positionId) {
  return request({
    url: `/api/positions/${positionId}`,
    method: 'delete'
  });
}
